spec/schemas/_common.query_dsl.yaml

```diff
@@ -69,11 +69,16 @@ components:
           description: |-
             Returns documents that match a provided text, number, date or boolean value.
             The provided text is analyzed before matching.
-          type: object
-          additionalProperties:
-            $ref: '#/components/schemas/MatchQuery'
-          minProperties: 1
-          maxProperties: 1
+          anyOf:
+            - type: object
+              additionalProperties:
+                $ref: '#/components/schemas/MatchQuery'
+              minProperties: 1
+              maxProperties: 1
+            - type: object
+              additionalProperties: true
+              minProperties: 1
+              maxProperties: 1
```

Comment:
doesn't the later (`additionalProperties: true`)  already encapsulate the former? or is it a placeholder for when we can accurately define another format for this?

Reply:
We have an `additionalProperties: true` and 

```
additionalProperties:
   type: object
```

This is quite different. The first one allows you to do `match: x`, and the second

```
match:
  x: y
```

Reply:
I'm curious if the following is allowed in JSON schema.
```yaml
type: object
additionalProperties:
  anyOf:
     - true
     - $ref: '#/components/schemas/MatchQuery'
minProperties: 1
maxProperties: 1
```

Reply:
Just tried it with the new JsonSchemaValidator:
```js
validator.validate_schema({ additionalProperties: { anyOf: [{type: 'string'}, { $ref: '#something'}]} })
```
It's allowed :D

Reply:
Indeed! `is_ref` needed some tweaking, https://github.com/opensearch-project/opensearch-api-specification/pull/405/commits/922d1543a49243c87bd7058378602df5aebaf43d